### ISSUE TITLE
Can't render PGraphics object using image() within a PDF

### ISSUE THREAD

**Issue by alignedleft** [processing/processing#4473]
@almoce logged this issue here: https://github.com/processing/processing-docs/issues/431

If you run this example, an error is triggered:

> ClassCastException sun.java2d.SunGraphics2D cannot be cast to java.awt.Image

Is this a bug? Shouldn't you be able to use image() to display/output a PGraphics object, even within the PDF renderer? (Or perhaps this isn't supported for PDFs.)

```
import processing.pdf.*;

PGraphics pg;

void setup() {
  size(100, 100, PDF, "output.pdf");
  pg = createGraphics(100, 100);
  noLoop();
}

void draw() {
  pg.beginDraw();
  pg.background(100);
  pg.stroke(255);
  pg.line(0, 0, 100, 100);
  pg.endDraw();

  image(pg, 0, 0);
}
```

**Comment by benfry** [comment]
That's really odd. It should work, and the error it's throwing is strange as well. Will have to look into it.

**Comment by lwaxman** [comment]
Hey Ben & Scott, just noting I'm having this issue as well. Trying to find a work-around, but not getting anywhere. Anyway, thanks for all your hard work.

**Comment by benfry** [comment]
Found and fixed for 3.2.
